components/turret.py

```diff
@@ -1,42 +1,101 @@
-class Turret:
-    # GEAR_RATIO: float
-    # ANGLE_ERROR_TOLERANCE: float
+from math import pi, radians, tau
+from rev import CANSparkMax
+from wpilib import DigitalInput
+from wpimath.controller import ProfiledPIDControllerRadians
+from wpimath.trajectory import TrapezoidProfileRadians
+from ids import SparkMaxIds, DioChannels
+
+from magicbot import tunable, feedback
+
 
-    # motor: Neo550 #This will need to change to the proper motor controller type
-    # positive_limit_switch: Digital I/O
-    # negative_limit_switch: Digital I/O # These could also be attached directly to the motor controller breakout board for interrupts
+GEAR_RATIO: float = (7 / 1) * (4 / 1) * (18 / 140)
+ANGLE_ERROR_TOLERANCE: float = radians(1)
+MAX_ANGULAR_VELOCITY: float = 1.0
+MAX_ANGULAR_ACCELERATION: float = 0.5
+NEGATIVE_LIMIT_ANGLE: float = radians(-120)
+POSITIVE_LIMIT_ANGLE: float = radians(120)
+INDEX_SEARCH_VOLTAGE: float = 2.0
+POSITIVE_SOFT_LIMIT_ANGLE: float = POSITIVE_LIMIT_ANGLE - radians(5)
+NEGATIVE_SOFT_LIMIT_ANGLE: float = NEGATIVE_LIMIT_ANGLE + radians(5)
 
-    # goal_angle = 0.0
+
+class Turret:
+    goal_angle = tunable(0.0)
 
     def __init__(self) -> None:
         # Create the hardware object handles
-
+        self.motor = CANSparkMax(
+            SparkMaxIds.turret_motor, CANSparkMax.MotorType.kBrushless
+        )  # This will need to change to the proper motor controller type
+        self.motor.setIdleMode(CANSparkMax.IdleMode.kBrake)
+        self.motor.setInverted(False)
+        self.positive_limit_switch = DigitalInput(DioChannels.positive_turret_switch)
+        self.negative_limit_switch = DigitalInput(
+            DioChannels.negative_turret_switch
+        )  # These could also be attached directly to the motor controller breakout board for interrupts
+        self.encoder = self.motor.getEncoder()
+        self.encoder.setPositionConversionFactor(GEAR_RATIO * tau)
+        self.encoder.setVelocityConversionFactor(GEAR_RATIO * tau)
```

Comment:
This conversation factor would give radians per minute. Bit of an oddball unit; when we use SI units we should be using them consistently.